src/charm.py

```diff
@@ -259,7 +266,8 @@ def _create_approle_secret(self, role_id: str, secret_id: str) -> Secret:
 
         # The secret already exists, so we will update it and log a warning.
         logger.warning(
-            f"Secret with label `{VAULT_CHARM_APPROLE_SECRET_LABEL}` already exists. Is the charm already authorized?"
+            f"Secret with label `{VAULT_CHARM_APPROLE_SECRET_LABEL}` already exists.\
+            Is the charm already authorized?"
```

Comment:
This is going to add some weird whitespacing. I suggest you use multiple strings in this case.

```suggestion
            f"Secret with label `{VAULT_CHARM_APPROLE_SECRET_LABEL}` already exists."
            " Is the charm already authorized?"
```

Python will automatically concatenate them. Don't forget the space (" ") at the beginning of the second string.

That being said, I hate that ruff enforces line length within strings. Seems really silly to me. This seems like a case where an ignore is totally warranted.